cpp/src/hash/hashing.cu

```diff
@@ -716,6 +718,60 @@ std::unique_ptr<column> md5_hash(table_view const& input,
                              mr);
 }
 
+std::unique_ptr<column> serial_murmur_hash3_32(table_view const& input,
+                                               uint32_t seed,
+                                               rmm::mr::device_memory_resource* mr,
+                                               cudaStream_t stream)
+{
+  auto output = make_numeric_column(
+    data_type(type_id::INT32), input.num_rows(), mask_state::UNALLOCATED, stream, mr);
+
+  if (input.num_columns() == 0 || input.num_rows() == 0) { return output; }
+
+  bool const nullable     = has_nulls(input);
+  auto const device_input = table_device_view::create(input, stream);
+  auto output_view        = output->mutable_view();
+  auto d_hashes           = output_view.data<hash_value_type>();
+
+  if (nullable) {
+    thrust::for_each(
+      rmm::exec_policy(stream)->on(stream),
+      thrust::make_counting_iterator(0),
+      thrust::make_counting_iterator(input.num_rows()),
+      [d_hashes, device_input = *device_input, original_seed = seed] __device__(auto row_index) {
+        uint32_t hash_result = original_seed;
+        for (int col_index = 0; col_index < device_input.num_columns(); col_index++) {
+          hash_result = cudf::type_dispatcher(
+            device_input.column(col_index).type(),
+            element_hasher_with_seed<MurmurHash3_32, true>{hash_result, hash_result},
+            device_input.column(col_index),
+            row_index);
+        }
```

Comment:
This could be better done as a `thrust::reduce` (or `transform_reduce`) with a `thrust::seq` exec policy.

Reply:
Restructured to use `thrust::tabulate` and a lambda with a `thrust::reduce`